core/station/api/src/external_canister.rs

```diff
@@ -43,10 +40,24 @@ pub struct CreateExternalCanisterOperationInput {
     pub name: String,
     pub description: Option<String>,
     pub labels: Option<Vec<String>>,
-    pub permissions: ExternalCanisterPermissionsInput,
+    pub permissions: ExternalCanisterPermissionsCreateInput,
     pub request_policies: ExternalCanisterRequestPoliciesInput,
 }
 
+#[derive(CandidType, serde::Serialize, Deserialize, Debug, Clone)]
+pub struct ExternalCanisterPermissionsCreateInput {
```

Comment:
Why does this have to be defined from scratch?

Reply:
can you be more specific?

Reply:
In the candid spec, there's a type synonym:
```
// The create input type for setting the permissions for the external canister.
type ExternalCanisterPermissionsCreateInput = ExternalCanisterPermissions;
```
but here we define the type fully.

Reply:
i see, makes sense, i did the name on the impl which now uses `ExternalCanisterPermissionsDTO`